[CS2113-W10-3] TraceYourProj

NgManSing opened the pull request:
TraceYourProj is a desktop app for tracking resources for data science projects, optimized for use via a Command Line Interface (CLI).

NgManSing committed:
[Sam] Updated UG
Fixes #94

NgManSing committed:
Merge branch 'master' into TP-UpdatedUG

gcksean committed:
Update Developer Guide for Save and Load functionality
Create new Sequence diagrams and update Developer guide with relevant info

yyixue committed:
Merge pull request #97 from jovanhuang/refactor-duke

Extract methods (Refactor) duke.java

yyixue committed:
Merge pull request #99 from seangoats/74-developer-guide-save-functionality

Update Developer Guide for Save and Load functionality

NgManSing committed:
[Sam] Minor fix on import class in Duke.java

yyixue committed:
Add find to help

gcksean committed:
Merge pull request #98 from NgManSing/TP-UpdatedUG

[Sam] Updated UG

yyixue committed:
Edit testListAllCommands

NgManSing committed:
[Sam] Updated DG

jovanhuang committed:
Add JUnit Test for List-all and list methods.

jovanhuang committed:
Merge pull request #102 from NgManSing/TP-UpdateDG

[Sam] Updated DG

jovanhuang committed:
Merge pull request #101 from yyixue/AddFindToHelp

Add find to help

jovanhuang committed:
Resolve checkStyleTest issue.

jovanhuang committed:
Refactor a bit.

gcksean committed:
Merge pull request #103 from jovanhuang/create-JUnit-test-for-list

Add JUnit Test for List-all and list methods.

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:
  Refactor a bit.
  Resolve checkStyleTest issue.
  Add JUnit Test for List-all and list methods.
  [Sam] Updated DG
  Edit testListAllCommands
  Add find to help
  [Sam] Minor fix on import class in Duke.java
  Update Developer Guide for Save and Load functionality Create new Sequence diagrams and update Developer guide with relevant info
  [Sam] Updated UG Fixes #94
  Extract methods (Refactor)
  refactor project class
  Added comments

jovanhuang committed:
add-sequence-diagram

jovanhuang committed:
Update list-all and list's feature to developer guide.

NgManSing committed:
Merge pull request #106 from jovanhuang/add-sequence-diagram

add-sequence-diagram

gcksean committed:
Merge pull request #108 from jovanhuang/add-sequence-diagram

Update list-all and list's feature to developer guide.

gcksean committed:
Update user guide for save and load

jovanhuang committed:
Update User Stories in Developer Guide (add list-all and list PROJECTNAME) stories

jovanhuang committed:
Merge pull request #109 from seangoats/75-update-user-guide

Update user guide for save and load

jovanhuang committed:
Update user guide for list-all and list PROJECTNAME features.

s-t-e-f committed:
Merge pull request #111 from jovanhuang/update-User-Stories-Developer-Guide

Update User Stories in Developer Guide (add list-all and list PROJECT NAME) stories

s-t-e-f committed:
Merge pull request #113 from jovanhuang/update-user-guide

Update user guide for list-all and list PROJECTNAME features.

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:
  Update user guide for list-all and list PROJECTNAME features.
  Update User Stories in Developer Guide (add list-all and list PROJECTNAME) stories
  Update user guide for save and load
  Update list-all and list's feature to developer guide.
  add-sequence-diagram

Zufiqqar commented on docs/DeveloperGuide.md:
```diff
@@ -19,8 +272,11 @@
 |v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
-
+|v2.0|user|find resources I have that are related to a keyword I specify|easily find resources I need without going through each project and resources
+|v2.0|user|see the list of resources for one of my project|recall what are the resources of one particular project in the database of TraceYourProj
```
![image](https://user-images.githubusercontent.com/57473144/113236835-439ebd80-92d8-11eb-90dc-9d8a21cbdc00.png)


Could be some bugs trying to display the actual table. Maybe the formating got something wrong

blank-bank reviewed (commented):
![image](https://user-images.githubusercontent.com/70243841/113381240-0b1ee280-93b1-11eb-826b-61003b1b7742.png)

The sequence diagram is overly complicated. Can consider extracting out the information into smaller diagrams by using the reference frames.

The colour usage  of the diagrams can be improve to enhance readability.
Adding notes can also help to explain the diagram.

blank-bank reviewed (commented):
![image](https://user-images.githubusercontent.com/70243841/113381542-cc3d5c80-93b1-11eb-9887-775f3e564201.png)

I think its uncompleted but can consider writing a short write up to explain the entire application and how it can be used before diving straight into the architecture. This will give the reader a clearer understanding of when the system is used.

blank-bank reviewed (commented):
![image](https://user-images.githubusercontent.com/70243841/113381741-4ff74900-93b2-11eb-9aa9-3d1a354ca955.png)

The text is very small. Can consider altering the size of the image using the img html tag.
https://stackoverflow.com/questions/14675913/changing-image-size-in-markdown

NgManSing committed:
[Sam] Fix bug about add command
- prompt user when the user is trying to add a resource with the same URL as the one that is already exsiting in the project's resource list.
- Updated the sequence diagram, DG, UG

s-t-e-f committed:
Merge pull request #115 from NgManSing/TP-HotFix-1

[Sam] Fix bug about add command

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:
  [Sam] Fix bug about add command - prompt user when the user is trying to add a resource with the same URL as the one that is already exsiting in the project's resource list. - Updated the sequence diagram, DG, UG

NoorSarrah commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,263 @@
+Aspect: How list-all executes
+* Alternative 1 (current choice):
+  * This list-all feature will do a loop through all the projects in the projects arraylist. For each project, it will loop through all the resources and print it out.
+* Alternative 2 (none)
```
![image](https://user-images.githubusercontent.com/77370767/113387629-432d2200-93bf-11eb-968b-8e88cffaebe1.png)
Should make the formatting more consistent.

NoorSarrah commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,263 @@
+**Step 3**:
+The user wishes to see the list of resources for one project, CZ2003. He executes the following command to see them.  
+
+>list *CZ2003*
```
Instead of > add p/IT3011 url/www.apple.com and >list *CZ2003*, you could use screenshots so that it stands out more.

NoorSarrah commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,263 @@
+The user can execute `list p/CS2113` to verify that the specified resource has been deleted from the project.
+
+The following sequence diagram shows how the delete operation works:
+![delete_puml](puml_img/deleteResource.png)
```
<img width="699" alt="dgReview2" src="https://user-images.githubusercontent.com/77370767/113388379-b5eacd00-93c0-11eb-89bb-240c5a3fb3d4.PNG">
Change this to a bigger diagram as the font is too small in the diagram and it affects readability

huachen24 replied:
I agree with Sarrah that this diagram is too wide and can't be read without zooming in.

NoorSarrah commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,263 @@
+>edit p/CS2113 i/2 url/www.CS2113tp.com d/This is tp site.
+
+**Step 3**:
+If PROJECT_NAME is not found in the database, the application prompts the user that the project is not found.
```
Should include screenshots of the possible outcomes to aid in the explanation.

Chiamjiaen reviewed (commented):
![image](https://user-images.githubusercontent.com/64964101/113426252-ca04ed80-9405-11eb-882e-1cdfe468f6b1.png)
Product Scope is incomplete although formatting seems to already be present

Chiamjiaen reviewed (commented):
![image](https://user-images.githubusercontent.com/64964101/113426455-118b7980-9406-11eb-83b9-eae11026436d.png)
good use of indentation in notes and design consideration also added but formatting seems to be off between alternative 1 and alternative 2

Chiamjiaen reviewed (commented):
![image](https://user-images.githubusercontent.com/64964101/113426591-44ce0880-9406-11eb-970d-4a5db7fd7608.png)
no indentation used in this section, making it very hard to read

Chiamjiaen reviewed (commented):
![image](https://user-images.githubusercontent.com/64964101/113426674-63340400-9406-11eb-8d93-3d2fee0a26bd.png)
This particular diagram is extremely small and hard to see

jovanhuang committed:
Update UserGuide.md

-Added outputs to all features.
-Beautify it.
-Fix minor bugs.

gcksean committed:
Merge pull request #118 from jovanhuang/update-UserGuide-Format

Update UserGuide.md

jovanhuang committed:
Fix minor bug in UserGuide.md

jovanhuang committed:
Update ppp.

jovanhuang committed:
Fix indentation bug in user guide

jovanhuang committed:
Update developer guide's title.

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:
  Update UserGuide.md

s-t-e-f committed:
Merge pull request #122 from jovanhuang/update-UserGuide-Format

Fix minor bug in UserGuide.md

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:
  Update developer guide's title.
  Fix indentation bug in user guide
  Update ppp.
  Fix minor bug in UserGuide.md

s-t-e-f committed:
Edited UserGuide and AboutUs

s-t-e-f committed:
Fix minor bug

huachen24 commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,263 @@
+
+* Alternative 2 (none)
+---
+### List all projects & their respective resources feature
```
Would it be better to standardize all your section headings? This has the word feature at the end but some of the other headings don't.

huachen24 commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,263 @@
+
+**Design Consideration**  
+
+Aspect: How list PROJECTNAME executes  
```
Can consider using proper command syntax notation (<> for required parameters, [] for optional parameters...)

huachen24 commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,263 @@
+> find k/website
+
+The following sequence diagram shows how the find operation works:
+![Exit_puml](puml_img/Find.png)
```
Try to use a lighter color for your background so that the contrast between the words and the background is higher to improve readability. You might also want to remove the :Model class

huachen24 commented on docs/DeveloperGuide.md:
```diff
@@ -1,9 +1,263 @@
+![Save_puml](puml_img/Save.png)
+
+The following sequence diagram shows how the **load** operation works:
+![Load_puml](puml_img/Load.png)
```
It would be good if you could explain what happens if there is no text file to read from.

yyixue committed:
Update user guide for find

huachen24 reviewed (commented):
The uml diagrams all have different colors and some include the Duke class while others do not. The way you show the inputs from the user are also a little different (execute("find ...") vs process command()). It would be good if all the diagrams could be standardized and a suitable color scheme be chosen.

yyixue committed:
Merge pull request #123 from s-t-e-f/master

Edited UG and AboutUs

jovanhuang committed:
Merge pull request #124 from yyixue/UpdateUserGuide

Update user guide for find

MingShun98 reviewed (commented):
The standardization of the headers could be better implemented. Difference in format may confuse the readers if the headers are within the same section.

![image](https://user-images.githubusercontent.com/56441669/113469789-0d517180-9483-11eb-9b06-e0127f4586ce.png)

MingShun98 commented:
Thought the content of the DG is specified, a table of content could be added at the start for better clarification of the different sections. Reading through a DG as such would be confusing to keep track. 

![image](https://user-images.githubusercontent.com/56441669/113469852-720ccc00-9483-11eb-9ee0-27ac87bff4c2.png)

MingShun98 commented:
Functionalities listed in the DG follows an implementation followed by numeric steps on the course of action. Could consider adding input and output examples to demonstrate the process on top of explanation.

MingShun98 commented:
Good effort with the addition of sequence diagrams to illustrate process. However, diagrams could be standardized to show a more cohesive DG instead of being done by separate individuals. In addition, the color of the attached diagrams is little difficult to read due the use of dark blue colors. Could consider following the lighter tone similar to above diagrams.

![image](https://user-images.githubusercontent.com/56441669/113469933-0b3be280-9484-11eb-93d1-472c939e499e.png)

jovanhuang committed:
Remove break blocks and replace them with return arrows and remove participants labels at the bottom.

jovanhuang committed:
Refactoring

jovanhuang committed:
Update Developer Guide.

gcksean committed:
Merge pull request #125 from jovanhuang/update-sequence-diag-list-feature

Update sequence diag list feature

jovanhuang committed:
Fix developer guide's format bug.

jovanhuang committed:
Update PPP

NgManSing committed:
Merge pull request #126 from jovanhuang/update-developer-guide-format

Fix developer guide's format bug.

NgManSing committed:
[Sam] Sequence Diagram update

gcksean committed:
Fix null description when loading project with no description

NgManSing committed:
Merge pull request #157 from seangoats/135-fix-null-description

Fix null description when loading project with no description

gcksean committed:
Fix bug where projects are appended if load is called multiple time

NgManSing committed:
[Sam] Bug Fix

1. [PE-D] Entering space character in the beginning causes app to crash: Fixed by triming the input and checking if command is null before switch in processCommand()
2. [PE-D] Order of parameters matters when adding a new resource: Fixed by adding another position check method
3. [PE-D] weird naming of variable in parser/CommandParser: Fixed by changing the naming of variable
4. [PE-D] Same project name but different projects will be created: Fixed by trimming the information
5. [PE-D] Project Name With Space In The End: Same as 4

Fixes #149
Fixes #153
Fixes #142
Fixes #127
Fixes #128

NgManSing committed:
[Sam] Fix checkstyle bugs

NgManSing committed:
[Sam] simplified the sequence diagram

NgManSing committed:
[Sam] cut the bottom of the Add.png

NgManSing committed:
[Sam] updated "Add" SD

NgManSing committed:
[Sam] update of add.png and my intro

NgManSing committed:
[Sam] Remove return/parameter types

yyixue committed:
Merge pull request #158 from seangoats/133-loading-same-projects-bug-fix

Fix bug where projects are appended if load is called multiple time

yyixue committed:
Merge pull request #159 from NgManSing/TP-HotFix-2

[Sam] Bug Fix

gcksean committed:
Fix load overwriting bug

Projects added before saving will be appended when load is called.
Override equals method in Project and Resource.
Add checking for empty projects so file is not overwritten to blank file.

yyixue committed:
Abstraction for Find

yyixue committed:
Add ProjectManager

yyixue committed:
Merge pull request #160 from NgManSing/TP-SequenceDiagram-1

Simplified "add" sequence diagram

yyixue committed:
Merge branch 'master' into 152-fix-load-overwriting

gcksean committed:
Fix conflict

yyixue committed:
Merge pull request #161 from seangoats/152-fix-load-overwriting

Fix load overwriting bug

gcksean committed:
Merge pull request #162 from yyixue/EditFindFunction

Fixed bugs with Find

gcksean committed:
Fix bug where project is not deleted

Project will now be deleted if delete entire project
or all resources are deleted.

yyixue committed:
Merge pull request #163 from seangoats/129-fix-project-cannot-delete

Fix bug where project is not deleted

yyixue committed:
Add finding for url of resource

yyixue committed:
Fix Checkstyle

gcksean committed:
Merge pull request #164 from yyixue/EditFindFunction

Add finding for url of resource

yyixue committed:
Simplify and edit find and exit uml diagrams

gcksean committed:
Merge pull request #165 from yyixue/CreatePlantUML

Simplify and edit find and exit uml diagrams

gcksean committed:
Add tests for Project and Resource and refactor packages

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp: (24 commits)
  Simplify and edit find and exit uml diagrams
  Fix Checkstyle
  Add finding for url of resource
  Fix bug where project is not deleted
  Fix conflict
  Add ProjectManager
  Abstraction for Find
  Fix load overwriting bug
  [Sam] Remove return/parameter types
  [Sam] update of add.png and my intro
  [Sam] updated "Add" SD
  [Sam] cut the bottom of the Add.png
  [Sam] simplified the sequence diagram
  [Sam] Fix checkstyle bugs
  [Sam] Bug Fix
  Fix bug where projects are appended if load is called multiple time
  Fix null description when loading project with no description
  [Sam] Sequence Diagram update
  Update PPP
  Fix developer guide's format bug.
  ...

s-t-e-f committed:
Fix bug for edit()

s-t-e-f committed:
Merge pull request #167 from seangoats/166-refactor-and-tests-resource-project

Add tests for Project and Resource and refactor packages

s-t-e-f committed:
Refractor codes.

s-t-e-f committed:
Fix checkstyle

s-t-e-f committed:
Fix checkstyle

s-t-e-f committed:
Fix checkstyle

s-t-e-f committed:
Fix JUnit tests

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:
  Add tests for Project and Resource and refactor packages

s-t-e-f committed:
no message

gcksean committed:
Create the first iteration of the class diagram.

Classes, methods and relations in this first iteration is based on current implementation details.
Update the DeveloperGuide to include the Class Diagram.

NgManSing committed:
[Sam] Fix bug related to command order and add junit tests

[PE-D] Certain wrong inputs manage to successfully create entry: Fixed
[PE-D] Invalid URLs accepted: Fixed, user could choose whether to perform URL check wihch requires internet access.

Fixes #137
Fixes #136

NgManSing committed:
[Sam] checkstyle stuff

gcksean committed:
Turn off icons for UML class diagram

gcksean committed:
Merge pull request #171 from NgManSing/TP-Junit-Add-1

[Sam] Fix bug related to command order and add junit tests

yyixue committed:
Merge pull request #168 from s-t-e-f/master

Fixes #146, #147

NgManSing committed:
[Sam] add unit tests

NgManSing committed:
[Sam] ..

NgManSing committed:
Merge pull request #170 from seangoats/169-create-starting-class-diagram

Create the first iteration of the class diagram.

NgManSing committed:
Merge branch 'master' into TP-Junit-Add-2

NgManSing committed:
[Sam] Refactor code to make it more coherent and added junit tests

gcksean committed:
Merge pull request #172 from NgManSing/TP-Junit-Add-2

[Sam] Refactor code to make it more coherent and added junit tests

yyixue committed:
Add to instructions for manual testing

gcksean committed:
Merge pull request #174 from yyixue/DG-InstructionsForManualTesting

Add to instructions for manual testing

yyixue committed:
Add JUnit Tests

yyixue committed:
Fix Checkstyle

yyixue committed:
Fix Checkstyle

gcksean committed:
Merge pull request #176 from yyixue/JUnitTestsFind

Add JUnit Tests

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:
  Fix Checkstyle
  Fix Checkstyle
  Add JUnit Tests
  Add to instructions for manual testing
  [Sam] Refactor code to make it more coherent and added junit tests
  [Sam] ..
  [Sam] add unit tests
  Turn off icons for UML class diagram
  [Sam] checkstyle stuff
  [Sam] Fix bug related to command order and add junit tests
  Create the first iteration of the class diagram.

s-t-e-f committed:
Added Javadocs

s-t-e-f committed:
Added Instructions for Manual Testing in developer guide.

s-t-e-f committed:
Updated sequence diagram for delete.

s-t-e-f committed:
Added sequence diagram for edit.

gcksean committed:
Merge pull request #177 from s-t-e-f/master

Added Instructions for Manual Testing

gcksean committed:
Update Class Diagram, DG and JavaDocs

Update Class diagram to be based on current implementation.
Add short intro to DG.
Add JavaDocs for Storage.

s-t-e-f committed:
Merge pull request #179 from seangoats/update-documentation

Update Class Diagram, DG and JavaDocs

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:
  Update Class Diagram, DG and JavaDocs

jovanhuang committed:
Update list feature to follow conventional format.

jovanhuang committed:
Fixed Minor runtest.sh bug.

gcksean committed:
Merge pull request #180 from jovanhuang/list-follow-conventional-format

Update list feature to follow conventional format.

jovanhuang committed:
Move list methods to Manager class.

jovanhuang committed:
Fix checkstyleMain bugs.

jovanhuang committed:
Fix checkstyleTest bug.

NgManSing committed:
Merge pull request #182 from jovanhuang/move-methods-to-manager-class

Move list methods to Manager class.

gcksean committed:
Fix: adding after loading

gcksean committed:
Fix load overwrite

NgManSing committed:
Merge pull request #184 from seangoats/small-fix

Fix: adding after loading

yyixue committed:
Add to ppp

jovanhuang committed:
Added exceptions handling to check for p/ in list feature.

jovanhuang committed:
Revert "Added exceptions handling to check for p/ in list feature."

This reverts commit 9aed4f1bbc01f65c58a3c43b18196990ebe8ec51.

jovanhuang committed:
Merge pull request #185 from yyixue/AddPPP

Add to ppp

gcksean committed:
Merge pull request #186 from jovanhuang/update-list-exceptions

Added exceptions handling to check for p/ in list feature.

jovanhuang committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

jovanhuang committed:
Re update list exceptions

gcksean committed:
Merge pull request #187 from jovanhuang/Reupdate-list-exception

Reupdate list exception

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:
  Re update list exceptions
  Revert "Added exceptions handling to check for p/ in list feature."
  Added exceptions handling to check for p/ in list feature.
  Add to ppp
  Fix load overwrite
  Fix: adding after loading
  Fix checkstyleTest bug.
  Fix checkstyleMain bugs.
  Move list methods to Manager class.
  Fixed Minor runtest.sh bug.
  Update list feature to follow conventional format.

gcksean committed:
Refactor: Load and Save and Update diags

s-t-e-f committed:
Merge pull request #188 from seangoats/update-seq-diag-load-save

Refactor: Load and Save and Update diags

s-t-e-f committed:
Added table of content for DG. Created anchors to headings in DG.

s-t-e-f committed:
Added draft PPP.

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:
  Refactor: Load and Save and Update diags

NgManSing committed:
Merge pull request #189 from s-t-e-f/master

Added Table of Content for DG and draft PPP.

s-t-e-f committed:
Updated PPP

gcksean committed:
Add PPP page

s-t-e-f committed:
Merge pull request #190 from seangoats/ppp-draft-sean

Add PPP page

s-t-e-f committed:
Added TOC for UG

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:
  Add PPP page

gcksean committed:
Merge pull request #191 from s-t-e-f/master

Updated PPP and added TOC for UG

jovanhuang committed:
Added more test cases, revamp exception classes for list & list-all features

jovanhuang committed:
Remove magic literals

jovanhuang committed:
Add author comments

jovanhuang committed:
Improve list-all sequence diagram.

jovanhuang committed:
Fix minor typos and bugs in DG.

jovanhuang committed:
Update list p/PROJECTNAME sequence diagram

jovanhuang committed:
Fix list sequence diagrams bugs

jovanhuang committed:
Fix checkStyleTest.

gcksean committed:
Merge pull request #192 from jovanhuang/remove-magic-literals-numbers

Remove magic literals numbers

jovanhuang committed:
Update sequence diagram names for list features.

yyixue committed:
Merge pull request #195 from jovanhuang/remove-magic-literals-numbers

Update the names of sequence diagrams for list features.

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:
  Update sequence diagram names for list features.
  Fix checkStyleTest.
  Fix list sequence diagrams bugs
  Update list p/PROJECTNAME sequence diagram
  Fix minor typos and bugs in DG.
  Improve list-all sequence diagram.
  Add author comments
  Remove magic literals
  Added more test cases, revamp exception classes for list & list-all features

gcksean committed:
Fix: delete not working after loading

jovanhuang committed:
Change error calling message method.

jovanhuang committed:
Remove author comments

jovanhuang committed:
Merge pull request #197 from seangoats/fix-delete-after-load

Fix: delete not working after loading

gcksean committed:
Merge pull request #198 from jovanhuang/Update-PPP

Change error calling message method.

jovanhuang committed:
Update sequence diagrams.

yyixue committed:
Merge pull request #204 from jovanhuang/update-sequence-diagram

Update sequence diagrams.

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

s-t-e-f committed:
Updated AboutUs

s-t-e-f committed:
Added JUnit Tests

s-t-e-f committed:
Fix checkstyle

gcksean committed:
Update Diagrams and Documentation

Create class diagram for Storage Component
Add Documentation in DG for Storage Component with class diagram
Standardize styles for Load and Save SDs

gcksean committed:
Add some logging and asserts

gcksean committed:
Merge pull request #205 from s-t-e-f/master

Added JUnit tests

gcksean committed:
Fix checkstyle

gcksean committed:
Update aboutus

yyixue committed:
Edit tests

yyixue committed:
Merge pull request #208 from seangoats/about-us-sean

Update aboutus

yyixue committed:
Merge pull request #206 from seangoats/class-diags

Update Diagrams and Documentation

yyixue committed:
Merge pull request #207 from seangoats/add-logging

Add some logging and asserts

yyixue committed:
Fix UML Diagrams

yyixue committed:
Fix AboutUs.md

gcksean committed:
Merge pull request #209 from yyixue/EditFindFunction

Edit tests

yyixue committed:
Add class diagram for ui and edit develoops guide

s-t-e-f committed:
Merge pull request #210 from yyixue/UpdateDevelopersGuide

Update developers guide and class diagram for find

s-t-e-f committed:
Edited sequence diagrams.

s-t-e-f committed:
Added more Junit tests and logging

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:
  Add class diagram for ui and edit develoops guide
  Fix AboutUs.md
  Fix UML Diagrams
  Edit tests
  Update aboutus
  Fix checkstyle
  Add some logging and asserts
  Update Diagrams and Documentation

# Conflicts:
#	src/main/java/seedu/duke/resource/ResourceManager.java

s-t-e-f committed:
Fix checkstyle

NgManSing committed:
[Sam] Updated DG, UG, Sequence Diagram of add, PPP, assert and added junit tests

s-t-e-f committed:
Fix checkstyle

NgManSing committed:
Merge branch 'master' into TP-AssertionLogging

# Conflicts:
#	docs/UserGuide.md
#	src/main/java/seedu/duke/project/ProjectManager.java
#	src/main/java/seedu/duke/storage/Storage.java

yyixue committed:
Added forgotten ui class diagram and update ppp

yyixue committed:
Merge pull request #211 from s-t-e-f/master

Edited sequence diagrams, added logging and more Junit tests

gcksean committed:
Merge pull request #212 from yyixue/AddPPP

Added forgotten ui class diagram and update ppp

jovanhuang committed:
Fix bug in DukeTest, testEditResource() and add loggings.

gcksean committed:
Merge pull request #213 from jovanhuang/add-logging

Fix bug in DukeTest, testEditResource() and add loggings for list features.

jovanhuang committed:
Add class diagrams for Model.

gcksean committed:
Merge pull request #214 from jovanhuang/add-model-class-diagram

Add class diagrams for Model.

jovanhuang committed:
Resolve minor bugs in userguide

jovanhuang committed:
Add jovanplaceholder to AboutUs

jovanhuang committed:
Update PPP for Jovan.

NgManSing committed:
[Sam] DG updates

NgManSing committed:
Merge branch 'master' into TP-AssertionLogging

# Conflicts:
#	docs/DeveloperGuide.md

NgManSing committed:
[Sam] fix some bugs

s-t-e-f committed:
Merge pull request #215 from jovanhuang/Update-PPP-Jovan

Update ppp jovan

NgManSing committed:
[Sam] Updated Architecture part

NgManSing committed:
[Sam] Updated Architecture part

NgManSing committed:
[Sam] fixed checkstyle promble

NgManSing committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp into TP-Meticulous-1

s-t-e-f committed:
Merge pull request #216 from NgManSing/TP-AssertionLogging

[Sam] updated the logging feature and documentation

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:
  [Sam] fixed checkstyle promble
  [Sam] Updated Architecture part
  [Sam] Updated Architecture part
  [Sam] fix some bugs
  [Sam] DG updates
  [Sam] Updated DG, UG, Sequence Diagram of add, PPP, assert and added junit tests

jovanhuang committed:
Fix bugs for DG.

gcksean committed:
Merge pull request #217 from jovanhuang/fix-bug-for-dg

Fix bugs for DG.

jovanhuang committed:
Remove return types and parameters in list-all & list p/PROJECT_NAME  sequence diagrams.

jovanhuang committed:
Fix minor bugs in jovan ppp

gcksean committed:
Merge pull request #218 from jovanhuang/keep-consistent-abstraction

Remove return types and parameters in list-all & list p/PROJECT_NAME …

gcksean committed:
Merge pull request #219 from jovanhuang/fix-jovan--ppp

Fix minor bugs in jovan ppp

yyixue committed:
Fix uml diagrams and about us

jovanhuang committed:
Merge pull request #220 from yyixue/UpdateDevelopersGuide

Fix uml diagrams and about us

NgManSing committed:
[Sam] Updates on Sequence diagram, PPP, Photo and other important stuffs

NgManSing committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp into TP-Meticulous-1

s-t-e-f committed:
Merge pull request #221 from NgManSing/TP-Meticulous-1

[Sam] Updates on PPP, diagram and other items

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:
  [Sam] Updates on Sequence diagram, PPP, Photo and other important stuffs
  Fix uml diagrams and about us
  Fix minor bugs in jovan ppp
  Remove return types and parameters in list-all & list p/PROJECT_NAME  sequence diagrams.
  Fix bugs for DG.

s-t-e-f committed:
Updated DG, UG and delete sequence diagram.

gcksean committed:
Merge pull request #223 from s-t-e-f/master

Edited DG, UG and delete sequence diagram.

s-t-e-f committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-3/tp:

s-t-e-f committed:
Added page breaks to UG, DG and PPP

s-t-e-f committed:
Set theme jekyll-theme-cayman

s-t-e-f committed:
Resize MainUI class diagram.

s-t-e-f committed:
Merge branch 'master' of https://github.com/s-t-e-f/tp

NgManSing committed:
Merge pull request #224 from s-t-e-f/master

Added page breaks and resize MainUI class diagram

NgManSing committed:
Update ngmansing.md

Minor update on styling

NgManSing committed:
Update ngmansing.md

Updated styling

NgManSing committed:
Update ngmansing.md

NgManSing committed:
Update ngmansing.md

NgManSing committed:
Update UserGuide.md

s-t-e-f committed:
Fix formatting issue when converting into pdf

NgManSing committed:
Update DeveloperGuide.md

NgManSing committed:
Update DeveloperGuide.md

s-t-e-f committed:
Standardise sequence diagrams

NgManSing committed:
Merge pull request #225 from s-t-e-f/master

Fix formatting issue when converting into pdf

NgManSing committed:
Update DeveloperGuide.md

NgManSing committed:
Update ngmansing.md

NgManSing committed:
Update ngmansing.md

NgManSing committed:
Update ngmansing.md

NgManSing committed:
Update ngmansing.md

NgManSing committed:
Update ngmansing.md

NgManSing committed:
Update ngmansing.md
